{"content": {"suites": {"0": {"status": {"total_pass": 0, "total_skip": 0, "total_xpass": 0, "total_xfail": 0, "total_rerun": 0, "total_fail": 3, "total_error": 6}, "tests": {"0": {"status": "ERROR", "message": "file /home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py, line 21\n  def test_TC_OD001_subscribes_to_scan_topic(node):\nE       fixture 'node' not found\n>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, extra, include_metadata_in_junit_xml, metadata, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, ros2_node, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory\n>       use 'pytest --fixtures [testpath]' for help on them.\n\n/home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py:21\n", "test_name": "test_TC_OD001_subscribes_to_scan_topic", "rerun": "0"}, "1": {"status": "ERROR", "message": "file /home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py, line 26\n  def test_TC_OD002_processes_ROI_data(node):\nE       fixture 'node' not found\n>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, extra, include_metadata_in_junit_xml, metadata, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, ros2_node, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory\n>       use 'pytest --fixtures [testpath]' for help on them.\n\n/home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py:26\n", "test_name": "test_TC_OD002_processes_ROI_data", "rerun": "0"}, "2": {"status": "ERROR", "message": "file /home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py, line 37\n  def test_TC_OD003_publishes_masked_scan(node):\nE       fixture 'node' not found\n>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, extra, include_metadata_in_junit_xml, metadata, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, ros2_node, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory\n>       use 'pytest --fixtures [testpath]' for help on them.\n\n/home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py:37\n", "test_name": "test_TC_OD003_publishes_masked_scan", "rerun": "0"}, "3": {"status": "ERROR", "message": "file /home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py, line 50\n  def test_TC_OD004_detects_obstacle_in_ROI(node):\nE       fixture 'node' not found\n>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, extra, include_metadata_in_junit_xml, metadata, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, ros2_node, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory\n>       use 'pytest --fixtures [testpath]' for help on them.\n\n/home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py:50\n", "test_name": "test_TC_OD004_detects_obstacle_in_ROI", "rerun": "0"}, "4": {"status": "ERROR", "message": "file /home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py, line 62\n  def test_TC_OD005_publishes_obstacle_status_true(node):\nE       fixture 'node' not found\n>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, extra, include_metadata_in_junit_xml, metadata, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, ros2_node, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory\n>       use 'pytest --fixtures [testpath]' for help on them.\n\n/home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py:62\n", "test_name": "test_TC_OD005_publishes_obstacle_status_true", "rerun": "0"}, "5": {"status": "FAIL", "message": "   AssertionError: assert 'Obstacle detected within 0.6m in the -90\u00b0 to -30\u00b0 range.' in ''\n    +  where '' = <_pytest.logging.LogCaptureFixture object at 0xffff799b3fa0>.text\n", "test_name": "test_TC_OD006_logs_warning_on_obstacle", "rerun": "0"}, "6": {"status": "FAIL", "message": "   AssertionError: assert 'No obstacles detected within 0.6m in the -90\u00b0 to -30\u00b0 range.' in ''\n    +  where '' = <_pytest.logging.LogCaptureFixture object at 0xffff744ecd90>.text\n", "test_name": "test_TC_OD007_logs_info_on_no_obstacles", "rerun": "0"}, "7": {"status": "ERROR", "message": "file /home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py, line 109\n  def test_TC_OD008_calculates_dynamic_indices(node):\nE       fixture 'node' not found\n>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, extra, include_metadata_in_junit_xml, metadata, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, ros2_node, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory\n>       use 'pytest --fixtures [testpath]' for help on them.\n\n/home/af/ros2_ws/src/ks_obstacle_detection/test/test_obstacle_detection.py:109\n", "test_name": "test_TC_OD008_calculates_dynamic_indices", "rerun": "0"}, "8": {"status": "FAIL", "message": "   AssertionError: assert 'Error processing LaserScan message' in ''\n    +  where '' = <_pytest.logging.LogCaptureFixture object at 0xffff744eca90>.text\n", "test_name": "test_TC_OD009_logs_errors_on_invalid_data", "rerun": "0"}}, "suite_name": "test/test_obstacle_detection.py"}}}, "date": "December 17, 2024", "start_time": 1734442561.554113, "total_suite": 1, "status": "FAIL", "status_list": {"pass": "0", "fail": "3", "skip": "0", "error": "6", "xpass": "0", "xfail": "0", "rerun": "0"}, "total_tests": "9"}